### PR TITLE
fix(design): fix linting warnings in design package

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@griest024 what do you think of the empty `toHaveBeenCalledWith`?

**Comment by griest024** [comment]
@damienwebdev I think there are times when you want to use `toHaveBeenCalled` when you don't care, or can't know, with what arguments a function is being called. However in general I prefer the stricter `toHaveBeenCalledWith`, especially for 0-arity functions.

File: libs/design/src/molecules/list/list/list.component.spec.ts [hunk 1]
```diff
@@ -109,14 +109,4 @@ describe('DaffListComponent', () => {
       });
     });
   });
-
-  describe('<daff-nav-list>', () => {
```
Comment (griest024) [inline libs/design/src/molecules/list/list/list.component.spec.ts:113]: why remove this test entirely?

Comment (xelaint) [reply libs/design/src/molecules/list/list/list.component.spec.ts:113]: It's a duplicate test.

Comment (xelaint) [reply libs/design/src/molecules/list/list/list.component.spec.ts:113]: https://www.shopgoodwill.com/Item/117857028

Comment (xelaint) [reply libs/design/src/molecules/list/list/list.component.spec.ts:113]: It's a duplicate test. It's already tested in line 67.

**Review by griest024 — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**